Code/GraphMol/MolPickler.cpp

```diff
@@ -810,6 +789,41 @@ void MolPickler::_pickle(const ROMol *mol, std::ostream &ss) {
       _pickleConformer<T>(ss, conf);
     }
   }
+
+  if(propertyFlags & PropertyPickleOptions::MolProps) {
+    streamWrite(ss, BEGINPROPS);
+    _pickleProperties(ss, *mol, propertyFlags);
+    streamWrite(ss, ENDPROPS);
+  }
+
+  if(propertyFlags & PropertyPickleOptions::AtomProps) {
+    streamWrite(ss, BEGINATOMPROPS);
+    for (ROMol::ConstAtomIterator atIt = mol->beginAtoms(); atIt != mol->endAtoms();
+         atIt++) {
+      _pickleProperties(ss, **atIt, propertyFlags);
+    }
+    streamWrite(ss, ENDPROPS);
+  }
+
+  if(propertyFlags & PropertyPickleOptions::BondProps) {
+    streamWrite(ss, BEGINBONDPROPS);
+    for (ROMol::ConstBondIterator bondIt = mol->beginBonds(); bondIt != mol->endBonds();
+         bondIt++) {
+      _pickleProperties(ss, **bondIt, propertyFlags);
+    }
+    streamWrite(ss, ENDPROPS);
+  }
+
+  if(propertyFlags & PropertyPickleOptions::QueryAtomData) {
+    streamWrite(ss, BEGINQUERYATOMDATA);
+    for (ROMol::ConstAtomIterator atIt = mol->beginAtoms(); atIt != mol->endAtoms();
+         atIt++) {
```

Comment:
Ug, I didn't notice this.  Seeing post-increment makes me throw up a little in my mouth.  I think I just copied the snippet from  a few lines below.  I'l fix it there as well.